### DISCUSSION TITLE
byte array []u8 should have a starts_with method

### DISCUSSION THREAD

**Discussion by David-Worboys** [vlang/v#15162] [Ideas]
byte array []u8 needs a starts_with and any relevant string methods that would map to a byte_array.  In this case I wrote my own

file_utils.bytes_starts_with(data_buffer, xmp_extended_id)

`

// Checks if if bstr_a starts wiith bstr_b
fn (fu File_Utils) bytes_starts_with(bstr_a []u8, bstr_b []u8) bool {
	if bstr_b.len > bstr_a.len {
		return false
	}

	for i in 0 .. bstr_b.len {
		if bstr_a[i] != bstr_b[i] {
			return false
		}
	}

	return true
}
`

**Comment by JalonSolov** [comment]
You need extra checks, such as `if bstr_a.len < bstr_b.len`, etc.

**Comment by David-Worboys** [comment]
Agree, mine was a quickie to get on with it. 

I still think byte_arrays should have these type of low level unctions built in so I do not have to roll my own

**Reply by JalonSolov** [reply]
Please feel free to improve your routine, then submit a PR.  Someone has to write the code, and contributions are always welcome.

**Comment by David-Worboys** [comment]
wilco! when I get a chance
